### PR TITLE
add is submitting logic to pages without any

### PR DESCRIPTION
**Pull request by petermason**
# Description

Prevent multiple clicks during transion between screens

## On what platform have you tested the change?

- [x] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

**Review by andyguest — COMMENTED** [review]
Peter, I was looking back to an earlier fix on the RegisterScreen - my evidence in the server logs suggests this fix may not actually be enough to prevent the double tap.  Can you do some local testing perhaps firing off toasts instead of sending reports to the backend to see if double hits remain possible?  

(It may be that I haven't identified the true source of the double taps though.)

https://github.com/zoe/covid-tracker-react-native/pull/1026/files

**Comment by petermason** [comment]
> Peter, I was looking back to an earlier fix on the RegisterScreen - my evidence in the server logs suggests this fix may not actually be enough to prevent the double tap. Can you do some local testing perhaps firing off toasts instead of sending reports to the backend to see if double hits remain possible?
> 
> (It may be that I haven't identified the true source of the double taps though.)
> 
> https://github.com/zoe/covid-tracker-react-native/pull/1026/files

I'm going to take another look at register and make a separate pr if required. This PR the logic is simpler, preventing additional taps after the first. This is achieved by changing a state variable which forces react to rerender the view, additional taps are blocked during this cycle. Once this cycle is complete the isSubmitting has been changed to true, thereofore, preventing additonal submissions. This state is only changed if the user returns to the page (back) or an error in the submission occurs.

**Review by bionara — APPROVED** [review]
Good looks good and is a logical fix to the double submit!

File: src/features/assessment/HowYouFeelScreen.tsx [hunk 1]
```diff
@@ -93,6 +100,10 @@ export const HowYouFeelScreen: React.FC<Props> = ({ route, navigation }) => {
 
   const currentPatient = assessmentCoordinator.assessmentData.patientData.patientState;
 
+  useEffect(() => {
```
Comment (selvach) [inline src/features/assessment/HowYouFeelScreen.tsx:103]: @petermason Sorry I didn't get to review this at the time. @petermason Could you explain what this does, please?

**Pull request merged**